pkg/loop/service.go

```diff
@@ -0,0 +1,154 @@
+package loop
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-plugin"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/logger"
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+var ErrPluginUnavailable = errors.New("plugin unavailable")
+
+const keepAliveTickDuration = 5 * time.Second //TODO from config
+
+var _ types.Service = (*ChainPluginService)(nil)
+var _ ChainPlugin = (*ChainPluginService)(nil)
+
+// ChainPluginService is a [types.Service] that implements [ChainPlugin], and maintains an internal client instance by
+// launching and re-launching as necessary using the provided [*plugin.ClientConfig].
+type ChainPluginService struct {
```

Comment:
This could live in core, but is useful as a testing tool for this repo and the chain specific ones.

Reply:
This is one of the reasons I *think* (but am not sure) we should have a single 'extensibility/plugins' repo where this is part of a 'chainplugin sdk', there will surely be other plugin types there in the future as well  - core can access it without needing relayers, relayers can access without needing core for testing purposes/tooling.

If we hate this idea, let's revisit post POC